pipelines/tekton/aiedge-e2e/tasks/model-registry-download.yaml

```diff
@@ -0,0 +1,115 @@
+          "$(params.model-registry-hostname)/api/model_registry/v1alpha3/model_versions/$MODEL_VERSION_ID" -H 'accept: application/json' | \
+          jq -r --arg MODEL_PARAM "fetch-model" '.customProperties[$MODEL_PARAM].string_value')
+          echo -n "$FETCH_MODEL" | tee $(results.fetch-model.path)
+
+          export GIT_MODEL_REPO=$(curl --silent -X 'GET' \
+          "$(params.model-registry-hostname)/api/model_registry/v1alpha3/model_versions/$MODEL_VERSION_ID" -H 'accept: application/json' | \
+          jq -r --arg MODEL_PARAM "git-model-repo" '.customProperties[$MODEL_PARAM].string_value')
+          echo -n "$GIT_MODEL_REPO" | tee $(results.git-model-repo.path)
+
+          export GIT_MODEL_REVISION=$(curl --silent -X 'GET' \
+          "$(params.model-registry-hostname)/api/model_registry/v1alpha3/model_versions/$MODEL_VERSION_ID" -H 'accept: application/json' | \
+          jq -r --arg MODEL_PARAM "git-model-revision" '.customProperties[$MODEL_PARAM].string_value')
+          echo -n "$GIT_MODEL_REVISION" | tee $(results.git-model-revision.path)
+
+          export MODEL_DIR=$(curl --silent -X 'GET' \
+          "$(params.model-registry-hostname)/api/model_registry/v1alpha3/model_versions/$MODEL_VERSION_ID" -H 'accept: application/json' | \
+          jq -r --arg MODEL_PARAM "model-dir" '.customProperties[$MODEL_PARAM].string_value')
+          echo -n "$MODEL_DIR" | tee $(results.model-dir.path)
+
+          export MODEL_RELATIVE_PATH=$(curl --silent -X 'GET' \
+          "$(params.model-registry-hostname)/api/model_registry/v1alpha3/model_versions/$MODEL_VERSION_ID" -H 'accept: application/json' | \
+          jq -r --arg MODEL_PARAM "modelRelativePath" '.customProperties[$MODEL_PARAM].string_value')
+          echo -n "$MODEL_RELATIVE_PATH" | tee $(results.modelRelativePath.path)
+
+          export S3-BUCKET-NAME=$(curl --silent -X 'GET' \
+          "$(params.model-registry-hostname)/api/model_registry/v1alpha3/model_versions/$MODEL_VERSION_ID" -H 'accept: application/json' | \
+          jq -r --arg MODEL_PARAM "s3-bucket-name" '.customProperties[$MODEL_PARAM].string_value')
+          echo -n "$S3-BUCKET-NAME" | tee $(results.s3-bucket-name.path)
+        fi
+        else
+          echo -n "Fetching params from PipelineRun"
+          echo -n "$(params.fetch-model)" | tee $(results.fetch-model.path)
+          echo -n "$(params.git-model-repo)" | tee $(results.git-model-repo.path)
+          echo -n "$(params.git-model-revision)" | tee $(results.git-model-revision.path)
+          echo -n "$(params.model-dir)" | tee $(results.model-dir.path)
+          echo -n "$(params.modelRelativePath)" | tee $(results.modelRelativePath.path)
+          echo -n "$(params.s3-bucket-name)" | tee $(results.s3-bucket-name.path)
+      fi  
+      
+  workspaces:
+  - description: The workspace for the downloaded model uris.
+    name: workspace
+  - description: The S3 secret.
+    name: s3-secret
```

Comment:
Are either of these workspaces actually used here?

Reply:
Sorry I missed this comment earlier. Good catch, actually this is not being used, I'm gonna test with this removed.

Reply:
@grdryn It turns out we need these workspaces and the pipeline fails without them with the below error: `workspace binding "s3-secret" does not match any declared workspace`

Reply:
I'm not suggesting that they be removed from the Pipeline altogether, just from these tasks, if they're not being used within this task. See the `check-model-and-containerfile-exists.yaml` task for example: the `s3-secret` workspace is neither used nor declared.